modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/extra/remote-app/src/routes/MDFRequestForm/steps/Activities/Activities.tsx

```diff
@@ -45,15 +45,32 @@ const Activities = ({
 	const [currentActivityIndex, setCurrentActivityIndex] = useState<number>(
 		values.activities.length
 	);
+	const [fromOnEdit, setFromOnEdit] = useState(false);
+
+	const [activityFormEmpty, setActivityFormEmpty] = useState(false);
 
 	const onAdd = () => {
 		setCurrentActivityIndex(values.activities.length);
+		setIsForm(true);
+		setFromOnEdit(false);
+
+		if (activityFormEmpty) {
+			arrayHelpers.remove(currentActivityIndex);
+			setActivityFormEmpty(true);
+		}
+	};
+
+	const onEdit = (index: number) => {
+		setCurrentActivityIndex(index);
+		setFromOnEdit(true);
 
 		setIsForm(true);
 	};
 
 	const onPreviousForm = () => {
-		arrayHelpers.remove(currentActivityIndex);
+		!fromOnEdit
+			? arrayHelpers.remove(currentActivityIndex)
+			: currentActivityIndex;
```

Comment:
This part using fromOnEdit and activityFormEmpty feels a little strange to me, I don't know if there is a better way to do it. But if it is working, it is fine with me.  I wish to hear others' opinions.